### DISCUSSION TITLE
Learning more about the region-based memory management

### DISCUSSION THREAD

**Discussion by bentxt** [melsman/mlkit#96] [Q&A] [answered]
Hi,

I try to learn more about the region based memory management functionality of mlkit. How and when it can be used. So while looking for hints in this codebase I stumpled upon this README in the 'test_dev' directory.
 I wonder if this could give some hints on how it is used and where its limitations are. 

test_dev/README, there is this note:

```
Programs that work with the KAM backend:

 a.pm                                 ok
 b.pm                                 ok
...
```

I read this term KAM (Kit Abstract Machine) somewhere in an older paper. Has this still some meaning? What does those example show excactly?

Ben

**Comment by bentxt** [comment]
small note: I think this is also interesting https://github.com/melsman/mlkit/pull/34

**Comment by melsman** [comment, marked as answer]
To get started with understanding the region based memory management scheme used in the MLKit, please consult [the mlkit manual](https://github.com/melsman/mlkit/raw/master/doc/mlkit.pdf). The `test_dev` folder is only relevant for compiler development and the `pm`-files are renamed into `mlb`-files. Also notice that the KAM backend (i.e., the bytecode backend) is no longer supported, which means that the KAM (Kit Abstract Machine) concept is purely conceptual.
